packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimeComponents/PickTimeClock.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from '../../../common/helpers';
+
+class PickTimeClock extends React.Component {
+  state = {
+    ampm: this.props.time.getHours() >= 12 ? 'PM' : 'AM'
+  };
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      ampm: nextProps.time.getHours() >= 12 ? 'PM' : 'AM'
+    };
+  }
+  setTime = (type, amount) => {
+    const { time } = this.props;
+    if (type === 'H') {
+      time.setHours(time.getHours() + amount);
+    } else if (type === 'M') {
+      time.setMinutes(time.getMinutes() + amount);
+    }
+    this.props.setSelected(time);
```

Comment:
can you add it to the destruction at the top of the method? 
I saw it also in other places over the code.